### DISCUSSION TITLE
Issues Advanced Search Private Beta

### DISCUSSION THREAD

**Discussion by rileybroughten** [community/community#137644] [Projects and Issues] [answered]
Thank you for participating in the [advanced search for Issues private beta](https://github.com/orgs/github/projects/4247/views/7?pane=issue&itemId=54825628), which comes alongside the [brand new Issues experience](https://github.com/orgs/community/discussions/132506), the [issue types private beta](https://github.com/orgs/community/discussions/112806), and the [sub-issues private beta](https://github.com/orgs/community/discussions/131957) ✨

Please note that right now this is a limited beta and as we scale it out further we will provide more information on how to join.

# Writing advanced searches

From the repository Issues page, you can now build advanced queries using the `AND` and `OR` keywords, as well as nested queries using parentheses up to 5 layers deep. Here is the [existing search syntax documentation](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests) for reference.

![repo index page](https://github.com/user-attachments/assets/b21bc930-9bce-49f5-a2ac-fedd3117db6d)

This search allows you to find issues that are either bugs _or_ have high priority.

```
is:issue AND ( type:Bug OR label:priority-high )
```

This search allows you to find issues that were created in the last week that are _either_ bugs or tasks, _or_ medium priority ones you've authored.

```
is:issue AND created:>@today-1w AND (( type:Bug OR type:Task ) OR ( author:@me AND label:priority-medium ))
```

Note: The `repo`, `org`, and `user` fields currently function as `OR` filters when a space is included between them, rather than an `AND` search, and are not yet supported in nested queries.

# Looking for feedback

Please leave your feedback below 👇  on what is working well, what can be improved, and what else you’d like to see!

**Comment by Dhrumil-Sentry** [comment]
Hello!

Some immediate feedback

I expect suggestions as soon as I click into the search bar
![image](https://github.com/user-attachments/assets/e7cc51bc-6c1d-42ba-875a-6ccf480ff48d)

I have to start typing and delete all text to see what I can search by
![image](https://github.com/user-attachments/assets/ab44f5aa-b126-4feb-bc23-5bb23f86994a)

My biggest gripe is that I need ways to search across the description or the comments of the issue - For e.g search for issues where users commented or search for issues with more than 10 emoji reactions or search for issues where the description contains "dashboards" - and it's still unclear as to how I'd do that

**Reply by rileybroughten** [reply]
> I expect suggestions as soon as I click into the search bar

Thank you for this feedback! Correct that right now you need to enter a character to see the suggestion dropdowns.

> My biggest gripe is that I need ways to search across the description or the comments of the issue

You can still type free text to match or can use the `in:title/body/comments` filter to search for text in specific places, such as `dashboard in:title`. Here is the [documentation](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests#search-by-the-title-body-or-comments). 

As you type out characters there will be additional filter suggestions to help guide your search, such as `comments` shown below.

<img width="1420" alt="filter input" src="https://github.com/user-attachments/assets/cfca70b3-2474-4f06-a0fe-8c4bcbe12b0a">

We're exploring ways we can continue to make the search experience more intuitive and easy to navigate, so this feedback is all very helpful!

**Comment by aatmmr** [comment]
Thaks a lot @rileybroughten! :) This will improve the search, and I look forward to playing with it. 

Something that comes to my mind immediately is that I would need to rewrite complex queries repeatedly. The feature [Saved Replies](https://docs.github.com/en/get-started/writing-on-github/working-with-saved-replies/using-saved-replies) allows the creation of reusable snippets for common answers on PRs and Issues. It would be helpful to have some sort of `Saved Issue Queries` that can be created somewhere and selected in the search field. It will indeed fail when a query parameter, e.g., "Type," does not exist, but there is validation that tells about failing requests.

I wonder if `Saved Issue Queries` would be better placed in a repository, as each repository has potentially its specific queries, or on a user level as [Saved Replies](https://docs.github.com/en/get-started/writing-on-github/working-with-saved-replies/using-saved-replies).

**Reply by rileybroughten** [reply]
Being able to save and reuse queries is definitely on our minds 😁 We've been thinking about saving queries on the user level. If you have any examples of common queries you'd like to save either on a user (or repository level if these would be helpful to share among a group of users) that would be helpful!

**Comment by Radiergummi** [comment]
While this sure looks nice, I think issue search is fundamentally lacking a way to search for unstructured text. When I'm facing an issue that's hard to pin down, I have to know the exact relevant keywords someone else might have encountered. If I don't, and the repository is active, I'll either find nothing, or hundreds of unrelated issues.  
This would be pretty much the *ideal* case for a vector search, possibly together with an LLM to translate the queries. I'm so weary of keyword hunting, or relying on Google to search GitHub for me.

I haven't found any existing discussions related to this (hah!), so I'm wondering if anyone at GitHub is working on something like that, too?

**Comment by beingminimal** [comment]
@rileybroughten Is this supported in projects v2? We need this there. Awaiting your response. Thanks

**Reply by rileybroughten** [reply]
This is not yet supported on Projects, just on the repository issues page currently. We'll provide updates when we expect this to be available in Projects 👍

**Comment by rileybroughten** [comment, marked as answer]
👋 Advanced search is now in [public preview](https://github.blog/changelog/2024-10-01-evolving-github-issues-public-preview/) 🎉! Moving forward, we will handle all opt-in requests via our [waitlist](https://github.com/features/issues/signup), so make sure to add your orgs there if you haven't already!

As we are moving out of private beta, we will also be closing this discussion in favor of our [public preview feedback thread](https://github.com/orgs/community/discussions/139934). Please direct all feedback there moving forward.
